File: Soap Bubble/Soap_BubbleView.m
//
//  Soap_BubbleView.m
//  Soap Bubble
//
//  Created by Felix Grabowski on 25/05/15.
//  Copyright (c) 2015 Felix Grabowski. All rights reserved.
//

#import "Soap_BubbleView.h"

@implementation Soap_BubbleView

- (instancetype)initWithFrame:(NSRect)frame isPreview:(BOOL)isPreview
{
    self = [super initWithFrame:frame isPreview:isPreview];
    if (self) {
        [self setAnimationTimeInterval:1/30.0];
        NSString *qtzComposition = [[NSBundle bundleForClass:[self class]] pathForResource:@"SoapBubble" ofType:@"qtz"];
        if (qtzComposition) {
            qtz = [[QCView alloc] initWithFrame: NSMakeRect(0, 0, NSWidth(frame), NSHeight(frame))];
            if (qtz) {
                [qtz loadCompositionFromFile:qtzComposition];
                [qtz setMaxRenderingFrameRate: 30.0f];
                [self addSubview:qtz];
            }
        }
    }
    return self;
}

- (void)startAnimation
{
    [qtz startRendering];
    [super startAnimation];
}

- (void)stopAnimation
{
    [qtz stopRendering];
    [super stopAnimation];
}

- (void)drawRect:(NSRect)rect
{
    [super drawRect:rect];
}

- (void)animateOneFrame
{
    return;
}

- (BOOL)hasConfigureSheet
{
    return NO;
}

- (NSWindow*)configureSheet
{
    return nil;
}

@end

File: Soap Bubble/SoapBubbleView.swift
//
//  File.swift
//  Soap Bubble
//
//  Created by Felix Grabowski on 26/05/15.
//  Copyright (c) 2015 Felix Grabowski. All rights reserved.
//

import ScreenSaver
import Quartz

var qtz:QCView?

class SoapBubbleView : ScreenSaverView {
    convenience init() {
        self.init(frame: CGRectZero, isPreview: false)
    }
    
    override init(frame: NSRect, isPreview: Bool) {
        super.init(frame: frame, isPreview: isPreview)
        setAnimationTimeInterval(1.0 / 30.0)
        var qtzComposition = NSBundle(forClass: self.dynamicType).pathForResource("SoapBubble", ofType: "qtz")
        if let comp = qtzComposition {
            qtz = QCView(frame: NSMakeRect(0, 0, NSWidth(frame), NSHeight(frame)))
            if qtz != nil {
                qtz!.loadCompositionFromFile(qtzComposition)
                qtz!.setMaxRenderingFrameRate(30.0)
                self.addSubview(qtz!)
            }
            
        }
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }
    
    
    override func startAnimation() {
        qtz!.startRendering()
        super.startAnimation()
    }
    
    override func stopAnimation() {
        qtz!.stopRendering()
        super.stopAnimation()
    }
    
    override func drawRect(rect: NSRect) {
        super.drawRect(rect)
    }
    
    override func animateOneFrame() {
        
    }
    
    override func hasConfigureSheet() -> Bool {
        return false
    }
    
    override func configureSheet() -> NSWindow? {
        return nil
    }
}
